### PR TITLE
Issue #69 - refactor current code base

### PR DESCRIPTION
**Pull request by psh**
In issue #69 (refactor current code base) it was suggested to merely break up the very large files into separate classes.  This PR goes a step beyond that suggestion to repackage by _component_ rather than _layer_, with common code the only exception to that rule.

image

Changes to the packaging resulted in minor changes in the sample app for imports that changed.

### CODE REVIEW

File: kakao/src/main/kotlin/com/agoda/kakao/common/matchers/ListViewViewAdapterSizeMatcher.kt [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import android.view.View
+import android.widget.ListView
+import org.hamcrest.Description
+
+/**
+ * Matches ListView with count of children
+ *
+ * @param size of children count in ListView
+ */
+class ListViewViewAdapterSizeMatcher(private val size: Int) : BoundedMatcher<View, ListView>(ListView::class.java) {
```
Comment (Unlimity) [inline kakao/src/main/kotlin/com/agoda/kakao/common/matchers/ListViewViewAdapterSizeMatcher.kt]: For some reason that I don't remember, the name of this class is wrong. It has double `View` in it. Could you please be so kind to change the name to `ListViewAdapterSizeMatcher`?

**Comment by cdsap** [comment]
Just reconfirm comments from @Unlimity, thank you very much for this great contribution!

**Comment by psh** [comment]
I've updated the PR with the requested changes.  Thanks!

**Comment by Unlimity** [comment]
Although, there are some problems with CI, we will check it as soon as we can, fix it, and them merge if all tests are passing.

**Comment by Unlimity** [comment]
@Vacxe could you please check and see what's wrong with CircleCI? Thanks in advance!

**Comment by Unlimity** [comment]
Everything passed! Great! Let's merge and I will prepare a full release within upcoming couple of weeks. Thanks again for your contribution!

**Pull request merged**